ocaml/xapi/export.ml

```diff
@@ -713,11 +713,15 @@ open Http
 open Client
 
 let lock_vm ~__context ~vm ~task_id op =
-  (* Note slight race here because we haven't got the master lock *)
-  Xapi_vm_lifecycle.assert_operation_valid ~__context ~self:vm ~op ~strict:true ;
-  (* ... small race lives here ... *)
-  Db.VM.add_to_current_operations ~__context ~self:vm ~key:task_id ~value:op ;
-  Xapi_vm_lifecycle.update_allowed_operations ~__context ~self:vm
+  Helpers.retry ~__context ~doc:task_id ~policy:Helpers.Policy.fail_quickly
```

Comment:
This appears to have some early wakeup code that will wake up the thread that is waiting whenever there is a change on the object class that caused the OTHER_OPERATION_IN_PROGRESS error, so should be fairly efficient.